src/components/common/TabPanel.tsx

```diff
@@ -40,55 +41,53 @@ function a11yProps(index: number) {
   };
 }
 
-export default function FullWidthTabs(props: {
+type FullWidthTabsProps = {
   children: React.ReactNode[];
   tabNames: string[];
   shouldPreventTabChange?: boolean;
   setShouldPreventTabChange?: FunctionType<void, boolean>;
-}) {
+};
+
+const FullWidthTabs: React.FC<FullWidthTabsProps> = ({
+  tabNames,
+  children,
+}) => {
   const theme = useTheme();
-  const [value, setValue] = React.useState(0);
+  const [query, setQuery] = useQueryParams({
+    tab: withDefault(NumberParam, 0),
+  });
 
   const handleChange = (
     event: React.ChangeEvent<Record<string, unknown>>,
     newValue: number
   ) => {
-    if (props.shouldPreventTabChange && value !== newValue) {
-      if (
-        window.confirm(
-          'Changes you recently made in this tab will be lost! Are you sure?'
```

Comment:
Is this removed because <PromptIfDirty /> is already taking care of confirmation?

Reply:
This is now handled by `<Prompt />` component since we are changing the url if you try to leave without saving the form. It is cleaner and nicer way of doing things. Previously I did a custom thing because when you change tabs nothing was changed in the url and I was checking manually if there is a change to prevent changing the tab.